### PR TITLE
Add private component and module for dynatrace private activegate

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
Add private component and module for dynatrace private activegate
- Deploy nonprod, will add prod at later stage

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: modules/dynatrace-private-activegate/cloudconfig.tpl [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#cloud-config
+package_upgrade: true
+
+runcmd:
+ - apt-get update
+ - apt-get -y install xvfb x11-xkb-utils xfonts-100dpi xfonts-75dpi xfonts-scalable
+ - apt-get -y install libasound2 libatk-bridge2.0-0 libatk1.0-0 libc6:amd64 libcairo2 libcups2 libgdk-pixbuf2.0-0 libgtk-3-0 libnspr4 libnss3 libxss1 xdg-utils
+ - curl --output chromium.tgz https://synthetic-packages.s3.amazonaws.com/Chromium/deb/chromium-89.0.4389.82-0ubuntu0.18.04.1.tgz
```
Comment (timja) [inline modules/dynatrace-private-activegate/cloudconfig.tpl:8]: should probably be a variable in the module?

File: modules/dynatrace-private-activegate/vm.tf [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    username   = local.adminuser
+    public_key = data.azurerm_key_vault_secret.ssh_public_key.value
+  }
+
+  custom_data = data.template_cloudinit_config.config.rendered
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "18.04-LTS"
```
Comment (timja) [inline modules/dynatrace-private-activegate/vm.tf:61]: there a reason we aren't using 20.04? 

they changed all the values in the offering for 20.04, see example:
https://github.com/jenkinsci/azure-vm-agents-plugin/blob/master/src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java#L1390

File: modules/dynatrace-private-activegate/vm.tf [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  gzip          = true
+  base64_encode = true
+
+  part {
+    content_type = "text/cloud-config"
+    content      = data.template_file.cloudconfig.rendered
+  }
+}
+
+resource "azurerm_linux_virtual_machine_scale_set" "main" {
```
Comment (timja) [inline modules/dynatrace-private-activegate/vm.tf:43]: there a reason we don't have log in with AAD enabled here? is it missing in the other active gate?

File: modules/dynatrace-private-activegate/vm.tf [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+
+data "template_file" "cloudconfig" {
+  template = file("${path.module}/cloudconfig.tpl")
+
+  vars = {
+    paas_token               = data.azurerm_key_vault_secret.dynatrace_paas_token.value
+    dynatrace_instance_name  = var.dynatrace_instance_name
+    network_zone             = var.network_zone
```
Comment (timja) [inline modules/dynatrace-private-activegate/vm.tf:29]: seems to be not defined?

File: components/dynatrace_activegate_private/nonprod.tf [hunk 5]
```diff
@@ -0,0 +1,16 @@
+module "nonprod_dynatrace_activegate_private" {
+  source                  = "../../modules/dynatrace-activegate_private"
+  instance_count          = 2
+  dynatrace_instance_name = "yrk32651"
+  env                     = "nonprod"
+  # network_zone            = "azure.cft"
```
Comment (timja) [inline components/dynatrace_activegate_private/nonprod.tf:6]: is there a network zone or not?

**Review by timja — APPROVED** [review]
Seems to be fine apart from the undefined network zone, few questions

File: modules/dynatrace-private-activegate/vm.tf (hunk 3 above)
Comment (thomast1906) [reply modules/dynatrace-private-activegate/vm.tf:43]: Yup, direct copy from the other active gate module

File: modules/dynatrace-private-activegate/vm.tf (hunk 2 above)
Comment (thomast1906) [reply modules/dynatrace-private-activegate/vm.tf:61]: Direct copy from the other active gate module, will upgrade this one to 20.04

File: components/dynatrace_activegate_private/nonprod.tf (hunk 5 above)
Comment (thomast1906) [reply components/dynatrace_activegate_private/nonprod.tf:6]: Want to test it without - will cleanup later, documentation didn't reference it but kept here just to confirm

File: modules/dynatrace-private-activegate/vm.tf (hunk 4 above)
Comment (thomast1906) [reply modules/dynatrace-private-activegate/vm.tf]: removed for now

File: modules/dynatrace-private-activegate/cloudconfig.tpl (hunk 1 above)
Comment (thomast1906) [reply modules/dynatrace-private-activegate/cloudconfig.tpl:8]: Will create variable if I go with the 1 module approach for both set of active gates

**Pull request merged**